modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,262 @@
+
+== Import a Type from a JSON Schema
+
+Having a JSON schema in the resources: 
+
+.JSON Script
+[source,JSON]
+----
+{
+  "$id": "https://example.com/person.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Person",
+  "type": "object",
+  "properties": {
+    "firstName": {
+      "type": "string",
+      "description": "The person's first name."
+    },
+    "lastName": {
+      "type": "string",
+      "description": "The person's last name."
+    },
+    "age": {
+      "description": "Age in years which must be equal to or greater than zero.",
+      "type": "integer",
+      "minimum": 0
+    }
+  },
+  "required": ["firstName", "lastName"]
+}
+----
+
+The following example shows how you can use the JSON schema loader to import the type:
+
+.DataWeave file header
+----
+import * from jsonschema!example::schema::Person
+----
+
+== Import Syntax
+
+In order to import the JSON schema types, as shown above, use the following directive:
+
+----
+import _typesToImport_ from jsonschema!_pathToJsonSchema_
+----
+
+The following example shows how to import a type:
+
+----
+import * from jsonschema!example::schema::Person
+----
+
+Where:
+
+* `typeToImport`: You can replace it by `\*` to import all types defined in the schema or to export a single type from the schema, for example, *Root*. You can also import JSON schema types with a different name, for example, *Root as Person*. This way you can reference the type with that name in the script.
+* `pathToJsonSchema`: This represents the path to the schema file, modifying the file separators for `::` and removing the “.json” extension from the file name. For example, if the path to the schema is “example/schema/Person.json”, “example::schema::Person” must be used.
+
+== Use Your Types in a DataWeave Script
+
+Having this directive in the script header loads all the existing types in the JSON schema. In this case, the only existing type is the *Root* type, specified at the root which describes an Object that can have three properties (`firstName`, `lastName`, and `age`). It is equivalent to declaring the following type in your DataWeave script:
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+type Root = { firstName: String, lastName: String, age?: Number }
+----
+
+Notice that the age field is the only one that is optional and it is signaled by the `?`.
+
+This is how you can use the type to know if a value follows the structure defined by the JSON schema.
+The following example outputs the value `true` since the object contains the required fields, `firstName` and `lastName`:
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ lastName: "Doe"
+} is Root
+----
+
+.Output
+----
+"True"
+----
+
+The following example outputs the value `false` since the object does not contain the required fields, `firstName` and `lastName`:
```

Comment:
Is the value `false` or `"False"`? I suspect it's `false` both here and in the output but can't test.

Reply:
Same as above, the value is: false .

Reply:
Replaced with `false`